mmv1/third_party/terraform/provider/provider.go.erb

```diff
@@ -406,6 +381,17 @@ func ProviderConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
 		return nil, diag.FromErr(err)
 	}
 
+	// Verify that universe domains match between credentials and configuration
+	if v, ok := d.GetOk("universe_domain"); ok {
```

Comment:
What does `ok` return on an explicit `""`? It's been a bit.

Provider values are weird compared to resource ones, and we may want to use `d.Get` for predictability.

Reply:
ok returns false on explicit "", so it'd be like it wasn't set at all. If credentials have a universe domain, users will get the error 
```
Universe domain mismatch: Universe domain 'X' was found in credentials without a corresponding 'universe_domain' provider configuration set. Please set 'universe_domain' to 'X' or use different credentials.
```
which seems appropriate?